UPGRADING.md

```diff
@@ -4,10 +4,116 @@ This document describes breaking changes and how to upgrade. For a complete list
 
 ## [Unreleased]
 
+## [2.0.0]
+
+This major release introduces several breaking changes and a redesigned API. While the codebase remains familiar, several design decisions were made to improve modularity, extensibility, and alignment with upstream Qiskit standards.
+The following sections describe the most important changes and how to adapt your code accordingly.
+We intend to provide a more comprehensive migration guide for future releases.
+
+### Migration to Qiskit's `Target` Class
+
+**Old:** MQT Bench-specific `Device` and `Gateset` classes.
+
+**New:** Qiskit's [`Target`](https://docs.quantum.ibm.com/api/qiskit/qiskit.transpiler.Target) class.
+
+So far, we used our own Device and Gateset classes to represent devices and native gatesets.
+In this release, we have switched to using Qiskit's `Target` class as the intermediate representation for both devices and native gatesets.
+This change allows us a more consistent and standardized way to represent quantum devices and their capabilities, modularity, and extensibility.
+
+Most previously used devices and gatesets are still available, but they are now provided as `Target` objects.
+You can retrieve the available names using:
+
+```python
+from mqt.bench.targets import get_available_gateset_names, get_available_device_names
+
+gateset_names = get_available_gateset_names()
+device_names = get_available_device_names()
+```
+
+Similarly, you can retrieve the corresponding `Target` objects by name:
+
+```python
+from mqt.bench.targets import get_target_for_gateset, get_device
+
+gateset = get_target_for_gateset("ibm_falcon", num_qubits=5)
+device = get_device("ibm_falcon_27")
+```
+
+This change allows you to use any device or gateset that is compatible with Qiskit's `Target` class, providing greater flexibility and compatibility with other Qiskit components.
+
+## Changes to the `get_benchmark` function
+
+The `get_benchmark` function has been redesigned to provide a more modular and extensible way to retrieve benchmarks.
+As a consequence, the function signature has changed.
+Please see the API documentation for the updated function signature and the examples provided below.
```

Comment:
This could use a link to the actual "API documentation", as this would make it more convenient for users.

Reply:
Done.